helper/scaleutils/nomad.go

```diff
@@ -83,6 +85,50 @@ func (si *ScaleIn) RunPreScaleInTasks(ctx context.Context, req *ScaleInReq) ([]N
 	return nodeIDMap, nil
 }
 
+// RunPostScaleInTasks runs any tasks that need to occur after a remote node
+// provider has completed its work. It handles any users configuration so that
+// the plugin does not need to perform this work.
+func (si *ScaleIn) RunPostScaleInTasks(cfg map[string]string, nodes []NodeID) error {
+
+	// Attempt to read of the node purge config parameter. If it has been set
+	// then check its value, otherwise the default stance is that node purging
+	// is disabled.
+	if val, ok := cfg[target.ConfigKeyNodePurge]; ok {
+
+		// Parse the string as a bool. If we get an error return this as the
+		// operator has attempted to configure this value.
+		boolVal, err := strconv.ParseBool(val)
+		if err != nil {
+			return fmt.Errorf("failed to parse %q: %v", target.ConfigKeyNodePurge, err)
```

Comment:
my only concern is how this error bubbles up... the alternatives seem to be a) pre-flight to check this config during start-up; b) treat invalid as a quiet(er) `false` (may not be much difference, aside from error metrics). i don't know whether i prefer either of these.

Reply:
I hadn't considered the impact on the error metrics. I think with this in mind I probably prefer the second option, with a log message indicating the failure to convert and then exiting. The pre-flight check, unless performed when the policy is read (which doesn't seem the correct place) can only happen when the plugin is called to perform the scale in action, at which point we would need to bail on the entire scale event, which to me seems less than ideal for a malformed boolean param controlling the very end of the pipeline. Nomad will also eventually perform this work if skipped anyway, so it's eventually consistent 😄  ?

Reply:
yeah, failure to purge isn't the end of the world.